### ISSUE TITLE
tabButton: misuse of refs causes tab conents to not appear when using separate content

### ISSUE THREAD

**Issue by elad661** [patternfly/patternfly-react#4095]
**Describe the issue. What is the expected and unexpected behavior?**
In our app (OCM), we use the <Tabs> component from PatternFly React (currently on     `"@patternfly/react-core": "^3.134.2"`).
Most of the tabs appear on mount, but some tabs take a few seconds before we know if we can show them or not.

We also use the option to have the tabs themselves and the tab contents in different components.

This causes an issue where in some cases, a tab that appears after mount won't be usable - switching to it shows an empty space where the tab contents should be. Further investigation showed the tab content `<section>` element had the `hidden` property, even tho the tab was selected.

Then I found out that this happens because the `tabContentRef` refers to the button, instead of the tab itself.

Looks like that in TabButton.tsx, line 22, 
https://github.com/patternfly/patternfly-react/blob/v4/packages/react-core/src/components/Tabs/TabButton.tsx#L22

Instead of reusing the name `tabContentRef`, you use just `ref` - this makes React allocate this ref to the TabButton itself if it wasn't allocated yet to the section.

**Please provide the steps to reproduce. Feel free to link CodeSandbox or another tool.**
This is a small reproducer with two tabs. The second tab appears 500ms after mount. Note how clicking on it doesn't switch between first/second tab contents.

https://codesandbox.io/s/thirsty-leftpad-yyz0b?file=/index.js

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**
While a workaround is theoretically possible, I haven't figured out one yet. Need to find a way to ensure the tab contents somehow claim the ref before the tab buttons...

**Comment by redallen** [comment]
> We also use the option to have the tabs themselves and the tab contents in different components.
...
> Need to find a way to ensure the tab contents somehow claim the ref before the tab buttons...

These conflict. We cannot control the order React assigns refs of sibling components. However, we can add a new `isHidden` prop to `<Tab>` which should allow you to hide/show tabs without needing the hacky `TabsRow` from your codesandbox.

**Comment by elad661** [comment]
> conflict. We cannot control the order React assigns refs of sibling components 

I know, that's why I listed this under the "workaround?" section. that is for "how will I work around this in my app", not how I suggest PF should work around it. I already explained the root cause of the issue  - you're using the world `ref` for this prop, even though this ref should refer to the tab content, not the button. That prop in TabButton shouldn't be called ref (because the ref being passed isn't supposed to refer to the tab button, it's supposed to refer to the tab contents ), and then PF won't grab an unnecessary ref.

With that said, the addition of `isHidden` is useful, so thanks for that.

I still think the underlying issue of the special prop name `ref` being used where it shouldn't be used.

**Comment by redallen** [comment]
> I already explained the root cause of the issue - you're using the world ref for this prop, even though this ref should refer to the tab content, not the button.

While you are correct the props passed to `TabButton` are labeled incorrectly, this is not the root cause of the issue. In fact we don't need to forward refs for `TabButton` at all, which is why that code you are referencing exists.

I spent an hour now investigating as to why we can't conditionally render `<Tab>` items in `<Tabs>`, and it's because we don't properly handle the cases of null children. I've opened #4211 to fix this which should allow `{showTab2 && <Tab />}` to behave like you expect.
